src/main/java/com/svydovets/bibirnate/cache/key/factory/KeyParamFactory.java

```diff
@@ -0,0 +1,52 @@
+package com.svydovets.bibirnate.cache.key.factory;
+
+import com.svydovets.bibirnate.cache.key.parameters.AbstractKeyParam;
+import com.svydovets.bibirnate.cache.key.parameters.EntityKeyParam;
+import com.svydovets.bibirnate.cache.key.parameters.QueryKeyParam;
+
+import java.util.Collection;
+import java.util.Objects;
+
+import static com.svydovets.bibirnate.cache.constant.CacheConstant.PARAMETER_CANNOT_BE_NULL;
+
+/**
+ * Provides overloaded methods for creating instances of the {@link AbstractKeyParam}.
+ */
+public class KeyParamFactory {
+
+    /**
+     * Provides instance of the {@link EntityKeyParam}. In case if some of passes parameters is null then throw
```

Comment:
either "in case" or "If" 
In case some passed parameters are ...
If some passed parameters are ...